src/regex.cr

```diff
@@ -274,15 +304,8 @@ class Regex
   def self.escape(str) : String
     String.build do |result|
       str.each_byte do |byte|
-        case byte.unsafe_chr
-        when ' ', '.', '\\', '+', '*', '?', '[',
-             '^', ']', '$', '(', ')', '{', '}',
-             '=', '!', '<', '>', '|', ':', '-'
-          result << '\\'
-          result.write_byte byte
-        else
-          result.write_byte byte
-        end
+        result << '\\' if needs_escape?(byte.unsafe_chr)
```

Comment:
Although the code is cleaner, beware that there might be performance penalty here.
Could some benchmark be done here to be sure what is the impact?

Reply:
Yup, I was thinking about that... Unfortunately there is some (~60%)

```
~ ❯ crystal run x.cr --release
Regex.escape (old)   2.24M (446.82ns) (± 9.54%)       fastest
Regex.escape (new)   1.41M (707.13ns) (± 5.37%)  1.58× slower
~ ❯ crystal run x.cr --release
Regex.escape (old)   2.39M ( 419.1ns) (± 6.82%)       fastest
Regex.escape (new)    1.4M (714.43ns) (± 6.30%)  1.70× slower
~ ❯ crystal run x.cr --release
Regex.escape (old)   2.23M (448.22ns) (± 6.34%)       fastest
Regex.escape (new)   1.43M (701.41ns) (± 4.63%)  1.56× slower
```

Reply:
Then, I prefer to rollback the change in the case statement. But put a comment in the case and in the constant declaration referencing each other.

Reply:
So be it... done.